src/player.ts

```diff
@@ -45,7 +45,7 @@ export default class Player extends EventEmitter {
       faction: this.faction,
       data: this.data,
       income: Reward.toString(Reward.merge([].concat(...this.events[Operator.Income].map(event => event.rewards))), true),
-      progress:  Object.values(finalScorings).forEach( (i) => ({ track: FinalTile[i], count: this.eventConditionCount(finalScorings[i])}))
+      progress:  Object.values(FinalTile).forEach( (i) => ({ track: FinalTile[i], count: this.eventConditionCount(finalScorings[i])}))
```

Comment:
As said, shouldn't it be `.map` instead of `.forEach`? Also `i` is very misleading, it should be named `tile` or `finalTile`.

Reply:
yes, `.map'. I'm working on it. I'm working on the satellites working. I understood is in the map. Right?

Reply:
The satellites is already done in `PlayerData.satellites` (see other comment).

Reply:
I'm sorry, but I don't have this.data.satellites

Reply:
https://github.com/donkeytech/gaia-engine/commit/e56670afe47b537aec62c3254ddd7019fa2480ad